apps/sbbtimetable/sbb_timetable.star

```diff
@@ -0,0 +1,203 @@
+                            font = FONT_TO_USE,
+                        ),
+                    ),
+                ] * NO_FRAMES_TOGGLE,
+            )
+
+            # Render the train delay
+            renderTimeNormal = render.Text(
+                content = "%s" % time.parse_time(trainTime, format = "2006-01-02 15:04:05").format("15:04"),
+                font = FONT_TO_USE,
+            )
+
+            if trainDelay == "+0":
+                renderTime = renderTimeNormal
+            else:
+                renderTimeChild = []
+                renderTimeChild.extend([renderTimeNormal] * NO_FRAMES_TOGGLE)
+                if trainDelay == "X":
+                    delayText = "--:--"
+                else:
+                    delayText = (time.parse_duration("%sm" % trainDelay) + time.parse_time(trainTime, format = "2006-01-02 15:04:05")).format("15:04")
+                renderTimeChild.extend(
+                    [
+                        render.Text(
+                            content = delayText,
+                            font = FONT_TO_USE,
+                            color = COLOR_DELAY,
+                        ),
+                    ] * NO_FRAMES_TOGGLE,
+                )
+                renderTime = render.Animation(children = renderTimeChild)
+
+            # Render the full row
+            childRow.append(
+                render.Row(
+                    children = [
+                        render.Animation(children = renderCategory),
+                        render.Box(width = 1, height = 5),
+                        renderTime,
+                        render.Text(
+                            content = "" if trainDelay == "+0" else "%s" % trainDelay,
+                            color = COLOR_DELAY,
+                            font = FONT_TO_USE,
+                        ),
+                        render.Marquee(
+                            width = 36 if trainDelay == "+0" else 36 - len(trainDelay) * 4,
+                            child = render.Text(
+                                content = "%s" % trainDest,
+                                font = FONT_TO_USE,
+                            ),
+                            offset_start = 0,
+                            offset_end = 0,
+                        ),
+                    ],
+                ),
+            )
+
+    return render.Root(
+        render.Column(
+            expanded = True,
+            main_align = "space_evenly",
+            cross_align = "center",
+            children = childRow,
+        ),
+    )
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "station",
+                name = "Station",
+                desc = "Station from which the timetable shall be shown.",
+                icon = "train",
+            ),
+            schema.Text(
+                id = "skiptime",
```

Comment:
Fancy! Effectively, how far away do you live from the train??

Reply:
Actually, I have a train station right next to my home. But on weekdays I use a train station to commute that is a 30 minutes car drive away.